flow/FileTraceLogWriter.cpp

```diff
@@ -48,6 +48,36 @@
 #include <fcntl.h>
 #include <cmath>
 
+struct IssuesListImpl {
+	IssuesListImpl(){}
+	void addIssue(std::string issue) {
+		MutexHolder h(mutex);
+		issues.insert(issue);
+	}
+
+	void retrieveIssues(std::set<std::string>& out) {
+		MutexHolder h(mutex);
+		for (auto const& i : issues) {
+			out.insert(i);
+		}
+	}
+
+	void resolveIssue(std::string issue) {
+		MutexHolder h(mutex);
+		issues.erase(issue);
+	}
+
+private:
+	Mutex mutex;
+	std::set<std::string> issues;
```

Comment:
This is kind of out of scope imho... This change would require some refactorying and I am not sure but I think it would also change existing behavior. So I won't make this change here